velox/connectors/hive/storage_adapters/hdfs/HdfsWriteFile.cpp

```diff
@@ -25,6 +25,13 @@ HdfsWriteFile::HdfsWriteFile(
     short replication,
     int blockSize)
     : hdfsClient_(hdfsClient), filePath_(path) {
+  auto pos = filePath_.rfind("/");
+  auto parentDir = filePath_.substr(0, pos + 1);
+  // Check whether the parentDir exist, create it if not exist.
+  if (hdfsExists(hdfsClient_, parentDir.c_str()) == -1) {
+    hdfsCreateDirectory(hdfsClient_, parentDir.c_str());
+  }
```

Comment:
does this work for "/example.parquet" ?

Reply:
Yes. This [ut ](https://github.com/oap-project/velox/blob/5b1806e99a5140ab6d35113dfc2da3eba237c694/velox/connectors/hive/storage_adapters/hdfs/tests/HdfsFileSystemTest.cpp#L381)can test this case.